### PR TITLE
Add Cloud Command Reference page

### PR DESCRIPTION
**Pull request by kate-goldenring**
Cloud commands

We will likely want to link to the cloud plugin blog from here. 

~Draft PR to see what folks think about having this added page in cloud to describe the `spin cloud` subcommands.~

~For some reason, the tabs are not rendering correctly :/~

~Note: Made up the existence of a cloud plugin `v0.1.1` that has an unhidden variables command. Likely we will actually replace this version~ removed

~TODO: add definition of what supported spin version for each cloud plugin version~

Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)
- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors
- [x] Relates to an existing (potentially outdated) blog article? If so please add URL in blog to point to this content.

### CODE REVIEW

**Comment by itowlson** [comment]
I think the tabs issue is they need `enable_shortcodes` in the page header.  (Not sure why this isn't the default.  I always forget it.)

**Comment by itowlson** [comment]
This structure seems good to me.  In the sidebar, I'd suggest renaming "CLI Reference" to "Spin CLI Reference" and moving "Cloud Command Reference" up one to be next to it.  But that's just minor tweaks.

**Comment by karthik2804** [comment]
@itowlson It is better not to have `enable_shortcodes` by default because it comes with an added cost to rendering.

File: content/cloud/cloud-plugin.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
+title = "Cloud Command Reference"
+date = "2023-06-07T22:41:02.478752Z"
+template = "cloud_main"
```
Comment (karthik2804) [inline content/cloud/cloud-plugin.md:3]:
```suggestion
template = "cloud_main"
enable_shortcodes = true
```

**Comment by kate-goldenring** [comment]
@karthik2804 are the tabs okay with `>=` sign? They seem to be rendering fine

**Comment by kate-goldenring** [comment]
Should we link to the cloud-plugin repo?

**Comment by karthik2804** [comment]
@kate-goldenring they render find and will consistently match across tabs if they are the same string. what would not work is if using something like `>=1.2` and then having another like `>=1.1`, `1.2`.
The above example would not break per say but auto switching of tabs would not work as expected.

**Comment by kate-goldenring** [comment]
@karthik2804 @itowlson @melissaklein24 I think this should be ready to review. I added the note about spin compatibility within each version tab instead of in the tab name. Also, instead of replacing the 0.1.0 plugin, we are putting out a 0.1.1 plugin so added tabs for it

File: content/cloud/cloud-plugin.md [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+  - [Login](#login)
+  - [Variables](#variables)
+    - [Set (Variables)](#set-variables)
+    - [Delete (Variables)](#delete-variables)
+    - [List (Variables)](#list-variables)
+  - [Subcommands Stability Table](#subcommands-stability-table)
+
+## Spin Cloud Command
+
+Fermyon provides a []`cloud` plugin](https://github.com/fermyon/cloud-plugin) for the Spin CLI for you to manage Spin applications in Fermyon Cloud. This page documents the `spin cloud` command. Specifically, all of the available options and subcommands. For more information on subcommand stability, see the [subcommands stability table](#subcommand-stability-table). You can reproduce the Spin Cloud command documentation on your machine by using the `--help` flag. For example:
```
Comment (itowlson) [inline content/cloud/cloud-plugin.md:20]: Is the extra `]` just before `cloud` (at the start of the para) intentional?

File: content/cloud/cloud-plugin.md [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+
+{{ tabs "cloud-plugin-version" }}
+
+{{ startTab "v0.1.0"}}
```
Comment (itowlson) [inline content/cloud/cloud-plugin.md]: As 0.1.1 will be the 'current' version at Spin 1.3 (which is itself the first time users can run `cloud` as a plugin), I'd suggest dropping the 0.1.0 tab.  Not a blocker if other folks want to keep it though!

**Comment by kate-goldenring** [comment]
>Also, instead of replacing the 0.1.0 plugin with unhidden variables commands, we are putting out a 0.1.1 plugin so added tabs for it

Chatted with @itowlson and decided to remove `v0.1.0` per his recommedation rather than add docs we dont assume anyone will use

**Review by itowlson — APPROVED** [review]
LGTM - one non-blocking suggestion (and one possible typo where there's no need to re-review if it does need a fix)

**Pull request merged**